microsoft/electionguard-api-python
# Feature/key ceremony

### Issue
*Link your PR to an issue*

Fixes #142 #143

### Description
Update to the latest version of the electionguard core repo.  Add state for the key ceremony.  Recreate, rework, and refactor api's to complete a key ceremony.

This PR also only includes the "happy path" meaning the challenge process is stubbed in but currently untested.

Also, of note, all of the request response objects currently return rich objects.  All of the data is available through the state changes but there are currently no "quality of life" endpoints to make fetching only relevant data easier (for instance, fetching just guardian backups rather than fetching the whole object)

### Testing
The only way to test this is to run it locally and manually execute all of the steps of the key ceremony, in order.  This is tedious and requires manually modifying requests and responses.  Working on a way to make it better and I will continue to update the PR for a better answer.  for now one can run the following steps, extracting the relevant data from the get requests and the response bodies to build up the data necessary for the next request.  The examples included in the postman collection are for Guardian_1's steps, but need to be modified to complete the process.

To Test:
1. load the postman tests and fire up docker-dev
2. mediator > Create Key Ceremony, Open Key Ceremony
3. mediator > Create Key Ceremony Guardian (repeat for all)
4. guardian > Create Guardian (repeat for all guardians)
5. guardian > Fetch Public Keys (repeat for all guardians) - make note of them
6. mediator > Announce Guardian (repeat for all)
6. mediator > Fetch Key Ceremony Guardian (repeat for all) - make note of them
7. guardian > Create Guardian Backup (repeat for all) - make note of them
8. mediator > Share Backups (repeat for all of them)
9. mediator > Fetch Key Ceremony Guardian (repeat for all) - make note of them
10. guardian > Verify Backup (repeat for all) - make note of them
11. mediator > Verify Backups (repeat for all)
12. mediator > Publish Joint Key
